### ISSUE TITLE
How did you choose hyperparameter for the vision task?

### ISSUE THREAD

**Issue by hzhz2020** [microsoft/Semi-supervised-learning#82]
Hi, 

thanks for the great work. 
For the reported performance on the cv task in your paper, how did you choose what parameter to use? Did you mention it in the paper, i couldn't find it.

Thanks!

**Comment by qianlanwyd** [comment]
Hi，we sweep the shared parameters such as learning rate based on FixMatch and choose the best one.

**Comment by Hhhhhhao** [comment]
> 

Additional validation split from training data is used for hyper-parameter search.

**Comment by hzhz2020** [comment]
In your paper you said 'validation data are provided to the cv datasets'?

<img width="975" alt="Screenshot 2023-03-01 at 2 46 10 PM" src="https://user-images.githubusercontent.com/32751718/222248729-afc543e3-120a-4b83-8efc-222b18d4a0b5.png">

**Comment by hzhz2020** [comment]
How many validation data did you use for TissueMNIST?

**Comment by Hhhhhhao** [comment]
TissueMNIST has a a validation set, so we use that.

https://github.com/microsoft/Semi-supervised-learning/blob/b979a23a1065f4d427350534a75ba1beb8c2afe9/semilearn/datasets/cv_datasets/medmnist.py#L237-L261

**Comment by hzhz2020** [comment]
thank you!

**Comment by hzhz2020** [comment]
Hi 
Sorry for opening this issue again. I read the paper again, and is still confused about your Table3 in the paper
<img width="998" alt="Screenshot 2023-05-02 at 8 27 29 PM" src="https://user-images.githubusercontent.com/32751718/235812615-9897746a-f776-4ffe-abfd-d76ea8f0c03a.png">

You replied above say that for TissueMNIST you use a validation set of size 23640 to select hyperparameters, but in Table3 you said **validation data are not provided for CV datasets**, and the corresponding field for TissueMNIST #Validation data is blank. 

How should i understand your previous reply and what's in the paper?

Thanks, and sorry for brining this up again.

**Comment by Hhhhhhao** [comment]
> Hi Sorry for opening this issue again. I read the paper again, and is still confused about your Table3 in the paper <img alt="Screenshot 2023-05-02 at 8 27 29 PM" width="998" src="https://user-images.githubusercontent.com/32751718/235812615-9897746a-f776-4ffe-abfd-d76ea8f0c03a.png">
> 
> You replied above say that for TissueMNIST you use a validation set of size 23640 to select hyperparameters, but in Table3 you said **validation data are not provided for CV datasets**, and the corresponding field for TissueMNIST #Validation data is blank.
> 
> How should i understand your previous reply and what's in the paper?
> 
> Thanks, and sorry for brining this up again.

Might be a typo in paper.  TissueMNist has a validation set so we use that instead splitting a small part of training data.

**Comment by hzhz2020** [comment]
now it makes sense! thank you!

**Comment by hzhz2020** [comment]
I also have a related question for the previous benchmark from your group, TorchSSL. For those benchmarking results in TorchSSL did you use a validation set? If so what is the size of the validation set?
